### DISCUSSION TITLE
Using Git Extensions for Collaboration Across Three Networked Computers

### DISCUSSION THREAD

**Discussion by ItzOlix** [gitextensions/gitextensions#12068] [General]
Hello guys,

Me and 2 of my friends are interns who will be working on a collaborative project using Git Extensions. All three of our computers are connected to the same local network. Could you please provide guidance on how to set up and use Git Extensions effectively in this environment? Specifically, we would like to know the following:

- How to configure a central repository on our network for shared access.
- Any best practices for collaboration to avoid conflicts and maintain smooth workflows.
- Additional tips or features in Git Extensions that might benefit our setup.
Thank you for your assistance!

Best regards,
Oliwier/ItzOlix
[Please if you can send response on olijab2007@gmail.com]

**Comment by mstv** [comment]
For small teams, a bare repo on a network share might suffice.
You can use all branching schemes.

If you want to establish a process with pull requests and CI or to use Git LFS, I would go for a hosted Git repo.

**Comment by namtab00** [comment]
It comes with ops overhead and infra requirements, but I would go with [self-hosted GitLab](https://about.gitlab.com/install/).

@ItzOlix GitExtensions is "just" a GUI client for git and imposes no limitations on the distributed nature of git.
If you are all interns, I presume in a company that already does sw development, you should ask your internal manager what (D)VCS they already use.

**Reply by pmiossec** [reply]
> It comes with ops overhead and infra requirements, but I would go with [self-hosted GitLab](https://about.gitlab.com/install/).

If you want to self host, an alternative maybe easier to setup: https://about.gitea.com/ or https://forgejo.org/ (fork of the 1st one)

But as a beginning, if you're only 3 and your repo is not big, a network share is enough the time to understand git. And decide later if you want/need self hosting or using an git hosting online service.

**Reply by RussKie** [reply]
Whilst a network share provides the git functionality, you'll be missing the pull-request review and other collaboration aspects, which are equally (if not more) important than the actual preservation of commit history.

**Comment by pmiossec** [comment]
>Any best practices for collaboration to avoid conflicts and maintain smooth workflows.

Globally, there are 2 workflow strategies : 'trunk base development' and 'branches development' (with here multiple possibilities : github flow, git flow,...)

It's too long to sum up with each advantages and inconvenients so I gave you the names so that you can find and read about it. You will more or less conflicts and not at the same time depending on the chosen strategy.
But the choice will depend on your maturity on a lot of subjects.
If it's a small starting project, trunk or github flow are good contenders.

Warning: this flow has the name of a gît hosting service github because they first introduced it but don't be confused in your search.
